### PR TITLE
feat: add performance metric to boot

### PR DESCRIPTION
**Pull request by capJavert**

### CODE REVIEW

File: src/telemetry/metrics.ts [hunk 1]
```diff
@@ -161,6 +182,16 @@ export const startMetrics = (serviceName: keyof typeof counterMap): void => {
...
       });
+
+      if (
+        requestDurationIncludePaths.includes(req.routeOptions.url as string)
+      ) {
+        requestDuration.record(res.elapsedTime, {
+          [ATTR_HTTP_REQUEST_METHOD]: req.method,
+          [ATTR_HTTP_RESPONSE_STATUS_CODE]: res.statusCode,
+          [ATTR_HTTP_ROUTE]: req.routeOptions.url,
+        });
```
[the remark this answers is not in the document]
Comment (capJavert) [reply src/telemetry/metrics.ts:193]: Ah nvm, request needs to happen so it activates, I see it
image

File: src/telemetry/metrics.ts [hunk 2]
```diff
@@ -149,10 +153,27 @@ export const startMetrics = (serviceName: keyof typeof counterMap): void => {
   const meterProvider = new metrics.MeterProvider({ resource, readers });
   api.metrics.setGlobalMeterProvider(meterProvider);
 
+  const performanceMeter = meterProvider.getMeter(`api-${serviceName}-perf`);
```
Comment (capJavert) [inline src/telemetry/metrics.ts:156]: Not sure if needs to be named something else?

Comment (capJavert) [reply src/telemetry/metrics.ts:156]: Just wanted to differentiate from other metrics, but maybe `serviceName` is also fine?

**Review by rebelchris — APPROVED** [review]
Looks good to me.
opentelemetry can abstract P50/P99 in their tooling?

Just wait for Ole-Martin as well though

**Comment by capJavert** [comment]
> Looks good to me.
> 
> opentelemetry can abstract P50/P99 in their tooling?
> 
> 
> 
> Just wait for Ole-Martin as well though

Yes I checked with Vas, will setup percentile graphs on google cloud side.

File: src/telemetry/metrics.ts (hunk 2 above)
Comment (omBratteng) [reply src/telemetry/metrics.ts:156]: You can still use the same meterProvider, it doesn't really make that much of a difference

**Review by omBratteng — APPROVED** [review]

**Pull request merged**